libsql-sqlite3/src/main.c

```diff
@@ -15,6 +15,7 @@
 ** accessed by users of the library.
 */
 #include "sqliteInt.h"
+#include "wal.h"
```

Comment:
We can't pull a large header here, SQLite was very strict with including only what's absolutely needed, and it's good practice, otherwise the dependencies become a huge mess. Normally you'd just forward-declare a few types here, like `typedef struct RefCountedWalManager RefCountedWalManager`, and then only use pointers, but I see that in one place, `libsql_open_v3`, you actually pass the whole `libsql_wal_manager` struct as value instead of as a pointer - why? It's rather unconventional to pass large structs by value, because they won't be passed via registers anyway. And passing a pointer in `libsql_open_v3` would give us the type erasure we need to avoid including the whole `wal.h` here.
Instead, you'd just `typedef libsql_wal_manager libsql_wal_manager` (like you already do in `sqlite.h.in`) and everything will work just fine. I very very highly recommend that, it's conventional in most C code, and even more conventional in our SQLite codebase.

Reply:
https://github.com/tursodatabase/libsql/pull/517#discussion_r1408253680